### PR TITLE
feat(signup-validations): Implement validations

### PR DESCRIPTION
**Pull request by huxaiphaer**
#### What does this PR do?
Implements signup validation.
#### Description of Task to be completed?
Validating username, email, and password when the user is registering to the application.
#### How should this be manually tested?
1. This can  be manually (locally) tested by using  this URL http://127.0.0.1:8000/api/users/ with a POST method in Postman, 
2. Then, you define your JSON body data as below to **REGISTER**.

```
{"user": {
                    "username": "lutaaya1" ,
                    "email": "lutaaya2@gmail.com",
                    "password": "12345554l"
          }
          
}
```
#### What are the relevant pivotal tracker stories?
[159952001](https://www.pivotaltracker.com/story/show/159952001)

### CODE REVIEW

**Comment by collinewait** [comment]
Great work though code climate shows some mentainablity issues shown bellow:

screen shot 2018-09-06 at 10 03 37

File: authors/apps/authentication/tests/test_login_register.py [hunk 1]
```diff
@@ -19,15 +19,15 @@ def setUp(self):
...
 
-    def test_api_can_register_a_user(self):
-        """Test the api can register a user."""
-        self.assertEqual(self.response.status_code, status.HTTP_201_CREATED)
-
     def test_api_can_login_a_user(self):
         """Test the api can login a user."""
         self.response = self.client.post(
             "/api/users/login/",
-            self.login_data,
```
Comment (collinewait) [inline authors/apps/authentication/tests/test_login_register.py]: Why was this line deleted?

File: authors/apps/authentication/tests/test_login_register.py [hunk 2]
```diff
@@ -19,15 +19,15 @@ def setUp(self):
             self.user_data,
             format="json")
 
-    def test_api_can_register_a_user(self):
```
Comment (collinewait) [inline authors/apps/authentication/tests/test_login_register.py]: There was no need of deleting and rewriting the same test. The deleted test was reusing data in the base test file instead of rewriting similar data in the test.

File: authors/settings/base.py [hunk 3]
```diff
@@ -20,7 +20,7 @@
 # See https://docs.djangoproject.com/en/1.11/howto/deployment/checklist/
 
 # SECURITY WARNING: keep the secret key used in production secret!
-SECRET_KEY = os.environ.get("SECRET_KEY", None)
+SECRET_KEY = os.environ.get("HOME", None)
```
Comment (collinewait) [inline authors/settings/base.py]: What is the reason behind replacing "SECRET_KEY" with "HOME"?

**Comment by collinewait** [comment]
Travis is failing.

File: authors/apps/authentication/tests/test_login_register.py (hunk 2 above)
Comment (huxaiphaer) [reply authors/apps/authentication/tests/test_login_register.py]: i have solved this @collinewait

**Comment by reiosantos** [comment]
I'm seeing  11 commits in your PR. could you please squash them together

**Comment by huxaiphaer** [comment]
@reiosantos  , am on it

File: authors/apps/authentication/tests/base_test.py [hunk 4]
```diff
@@ -29,4 +28,10 @@ def __init__(self):
                                           "password": self.password,
                                           }
                                  }
+
+        self.no_password_login = {
+            "user": {"email": self.user_email,
+                     "password": None,
```
Comment (inno-asiimwe) [inline authors/apps/authentication/tests/base_test.py:34]: Do we really need to pass none here?

Comment (huxaiphaer) [reply authors/apps/authentication/tests/base_test.py:34]: I was still figuring out the way to get the validation error,
